frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/moderation/ModerationPortletUtils.java

```diff
@@ -82,55 +92,96 @@ public static RequestStatus addCommentToClearingRequest(PortletRequest request,
     }
 
-    public static RequestStatus updateClearingRequest(PortletRequest request, Logger log) {
+    public static AddDocumentRequestSummary updateClearingRequest(PortletRequest request, Logger log) {
         String id = request.getParameter(PortalConstants.CLEARING_REQUEST_ID);
         User user = UserCacheHolder.getUserFromRequest(request);
         String isReOpen = request.getParameter(PortalConstants.RE_OPEN_REQUEST);
-        if (CommonUtils.isNotNullEmptyOrWhitespace(isReOpen) && Boolean.parseBoolean(isReOpen)) {
-            return reOpenClearingRequest(id, request, user);
-        }
-        String agreedDate = request.getParameter(ClearingRequest._Fields.AGREED_CLEARING_DATE.toString());
-        String status = request.getParameter(ClearingRequest._Fields.CLEARING_STATE.toString());
+        AddDocumentRequestSummary requestSummary = new AddDocumentRequestSummary().setRequestStatus(AddDocumentRequestStatus.FAILURE);
         if (null != id) {
+            if (CommonUtils.isNotNullEmptyOrWhitespace(isReOpen) && Boolean.parseBoolean(isReOpen)) {
+                return reOpenClearingRequest(id, request, user);
+            }
             try {
+                String isClearingExpertEdit = request.getParameter(PortalConstants.IS_CLEARING_EXPERT);
                 ModerationService.Iface client = new ThriftClients().makeModerationClient();
                 ClearingRequest clearingRequest = client.getClearingRequestByIdForEdit(id, user);
-                if (CommonUtils.isNotNullEmptyOrWhitespace(agreedDate) && !SW360Utils.isValidDate(agreedDate, DateTimeFormatter.ISO_LOCAL_DATE, 0)) {
-                    log.warn("Invalid agreed clearing date: " + agreedDate + " is entered, by user: "+ user.getEmail());
-                    return RequestStatus.FAILURE;
+                String clearingTeam = request.getParameter(ClearingRequest._Fields.CLEARING_TEAM.toString());
+                if (CommonUtils.isNullEmptyOrWhitespace(clearingTeam)) {
+                    log.warn("Invalid clearingTeam email: " + clearingTeam + " is entered, by user: "+ user.getEmail());
+                    return requestSummary.setMessage("Invalid clearingTeam email");
+                }
+                clearingRequest.setClearingTeam(clearingTeam);
+                if (CommonUtils.isNotNullEmptyOrWhitespace(isClearingExpertEdit) && Boolean.parseBoolean(isClearingExpertEdit)) {
+                    String agreedDate = request.getParameter(ClearingRequest._Fields.AGREED_CLEARING_DATE.toString());
+                    String status = request.getParameter(ClearingRequest._Fields.CLEARING_STATE.toString());
+                    String priority = request.getParameter(ClearingRequest._Fields.PRIORITY.toString());
+                    if (CommonUtils.isNotNullEmptyOrWhitespace(agreedDate) && !agreedDate.equals(clearingRequest.getAgreedClearingDate())
+                            && !SW360Utils.isValidDate(agreedDate, DateTimeFormatter.ISO_LOCAL_DATE, 0)) {
```

Comment:
AgreedClearingDate, can it be null? if yes can you please add a null check similar like done for agreedDate.

Reply:
No method is being invoked on `clearingRequest.getAgreedClearingDate()`, So there won't be any `nullPointerException` even if `agreedClearingDate` is null.